### PR TITLE
User id to username

### PR DESCRIPTION
**Pull request by OsvaldoPrajitno**
mainly modified the leaderboard endpoint to convert userid to username, instead of using username from the game object directly assuming we store userid in the game object.
(this branch is closely related to the getlocation branch)

### CODE REVIEW

File: backend/src/index.ts [hunk 1]
```diff
@@ -61,6 +61,15 @@ const session_remove = async (sessionId: string) => {
   return true;
 };
 
+const getUsername = async (userId: string) => {
+  const sessionData = query(users, where("__name__", "==", userId));
```
Comment (lachlanshoesmith) [inline backend/src/index.ts]: this var name doesn't seem to make a ton of sense

File: backend/src/index.ts [hunk 2]
```diff
@@ -61,6 +61,15 @@ const session_remove = async (sessionId: string) => {
...
+  const session = await getDocs(sessionData);
```
Comment (lachlanshoesmith) [inline backend/src/index.ts]: like above, doesn't seem to pertain to a session?

**Review by lachlanshoesmith — CHANGES_REQUESTED** [review]

File: backend/src/index.ts [hunk 3]
```diff
@@ -61,6 +61,15 @@ const session_remove = async (sessionId: string) => {
   return true;
 };
 
+const getUsername = async (userId: string) => {
+  const userData = query(users, where("__name__", "==", userId));
+  const user = await getDocs(userData);
+
+  if (user.empty) return 'user not found';
```
Comment (lachlanshoesmith) [inline backend/src/index.ts]: i think returning `undefined` makes more sense here imo, means we can write something later (see: leaderboard/data) like `if (getUsername(etc)) { ... }` that throws an error if a username cant be found, cos that shouldnt happen give a proper userid

File: backend/src/index.ts [hunk 4]
```diff
@@ -239,7 +251,7 @@ app.get(
     for (let i = 0; i < increments && i + start < size; i++) {
```
Comment (lachlanshoesmith) [inline backend/src/index.ts]:
```js
const end = Math.min(start + increments, size);
for (let i = start; i < end; i++) {
```

bit more readable imo

**Review by lachlanshoesmith — CHANGES_REQUESTED** [review]
mostly very good just a couple nitpicks

File: backend/src/index.ts [hunk 5]
```diff
@@ -317,11 +329,16 @@ app.get(
     const pageCount = Math.ceil(size / increments);
 
     const data: ScoreEntry[] = [];
-
-    for (let i = 0; i < increments && i + start < size; i++) {
+    const end = Math.min(start + increments, size);
+    for (let i = start; i < end; i++) {
+      // if username is undefined, don't add to leaderboard
+      let username: string;
+      if (!(username = await getUsername(queryScoreSnapshot.docs[i + start].data().userid))) {
```
Comment (bengodw) [inline backend/src/index.ts]: shouldn't this throw an error or something

Comment (OsvaldoPrajitno) [reply backend/src/index.ts]: something like  `throw error("Invalid userid in games")` ?

**Review by bengodw — DISMISSED** [review]

**Pull request merged**